bookshelf/bookshelf_object.py

```diff
@@ -0,0 +1,153 @@
+
+        Parameters
+        ----------
+        author: string, required
+            Author of new added book.
+        title: string, required
+            Title of new added book.
+
+        Returns
+        -------
+        Returns True if book was succesfull added or False on failure.
+
+        Example
+        -------
+        >>> my_bookshelf.add_book("Aldous Huxley", "Brave new world")
+        """
+
+        self.books[author] = title
+
+        if self.books[author] == title:
+            self.currentBookNumber += 1
+            return True
+        else:
+            return False
+
+    def remove_book(self, author: str) -> bool:
+        """
+        Remove book from books dictionary by key which is author of the book.
+
+        Parameters
+        ----------
+        author: sting, required
+            Name of the book author.
+
+        Returns
+        -------
+        Return True if book was successfully removed from books dictionary
+        or False on failure and when key author does not exist in books.
+
+        Example
+        -------
+        >>> my_bookshelf.remove_book("Aldous Huxley")
+        """
+
+        if self.books.pop(author, False):
+            self.currentBookNumber -= 1
+            return True
+        else:
+            return False
+
+    def rename(self, name: str) -> bool:
+        """
+        Change current name of Bookshelf instance.
+
+        Parameters
+        ----------
+        name: string, required
+            New name for Bookshelf instance.
+
+        Returns
+        -------
+        Return True if name was successfully change or False on failure.
+
+        Example
+        -------
+        >>> my_bookshelf.rename("My new books")
+        """
+
+        self.name = name
+
+        if self.name == name:
+            return True
+        else:
+            return False
+
+    def books_count(self) -> int:
+        """
+        Return amount of current books held in books dictionary.
+
+        Returns
+        -------
+        Return Bookshelf's instance attribute which holds number of current
+        stored books.
+
+        Example
+        -------
+        >>> my_bookshelf.book_count()
+        """
+
+        return self.currentBookNumber
+
+    def available_space(self) -> int:
+        """
+        Return amount of books that can be still held in books dictionary.
+
+        Returns
+        -------
+        Result of subtraction Bookshelf instance attributes maxSize and
+        currentBookNumber.
+
+        Example
+        -------
+        >>> my_bookshelf.avaiable_space()
+        """
+        return self.maxSize - self.currentBookNumber
+
+    def is_empty(self) -> bool:
+        """
+        Check if Bookshelf data structure books is empty.
+
+        Returns
+        -------
+        Return True if books dictionary is has 0 elements or False if has
+        more than 0 elements.
+        """
+
+        if self.books.__len__() == 0:
```

Comment:
Here is some critical issue.
This is probably some habit from C or C++?
Python is mostly dedicated to be readable. Usage of dunder methods as `__len__()` or `__eq__()` etc. are mostly reserved for some really internal implementations. When you need to decide what is better in implementation as `len()` or `__len__()`, 99% of the time it will be `len()`. Why? Python defines such user friendly methods just to use them :D It is more clear when you read it and more convenient for other programmers during some code modifications.

Comparing to C and C++, you have there `public` / `protected` / `private` methods or class variables.
These things are strictly defined by these languages.
Here in Python, we do not have such things. To be honest, everything is `public`. But...
There is a convention that methods or variables with one underscore suffix we treated as protected (we in the meaning of programmers and IDEs). Methods and variables with double underscore are treated as private.
So, we should stick for that convention and not break it. There is no reason to call a private `len()` method on `books`, as user also should be able to call it. There is no reason to hide it.

One possible suggestion:
```suggestion
        if len(self.books) == 0:
```

Or other, probably less readable, but you can check if dictionary is not empty:
```suggestion
        if not self.books:
```